src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -75,6 +77,35 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
+        public static Dictionary<string, int> FetchToolResultCounts(this SarifLog log)
+        {
+            if (log == null) { throw new NullReferenceException(); }
+            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+
+            Dictionary<string, int> resultCountsByTool = new Dictionary<string, int>();
+            int appropriateResultCount = 0;
+
+            foreach (Run run in log?.Runs)
```

Comment:
I feel like they're might be some Sarif trick here to make this faster, as it has to iterate through all the results to make sure we only count the one's we plan on filing.  I don't know it, though.

Reply:
Removing the filter makes this a simple grab of the count.

---
In reply to: [394674496](https://github.com/microsoft/sarif-sdk/pull/1817#discussion_r394674496) [](ancestors = 394674496)